### PR TITLE
fix: calculate log2i properly

### LINKED ISSUE #7080: clock skew
### Description

It looks like talos nodes aren't syncing their clocks properly. Ceph is complaining of clock skew and I can confirm this is the case on the hosts themselves. The difference between `2023-04-12 11:43:52.297503558 +0000 UTC` and `2023-04-12 11:43:52.430898682 +0000 UTC` is `133.395124ms`(!). ([playground](https://go.dev/play/p/Ba-CnUyot1r))

### Logs

```sh
❯ talosctl --talosconfig=unwind-talosconfig -n 10.0.0.101,10.0.0.102,10.0.0.103,10.0.0.104,10.0.0.105 time
NODE         NTP-SERVER     NODE-TIME                                 NTP-SERVER-TIME
10.0.0.101   pool.ntp.org   2023-04-12 11:43:52.324593578 +0000 UTC   2023-04-12 11:43:52.308599071 +0000 UTC
10.0.0.102   pool.ntp.org   2023-04-12 11:43:52.430898682 +0000 UTC   2023-04-12 11:43:52.303693959 +0000 UTC
10.0.0.103   pool.ntp.org   2023-04-12 11:43:52.317368811 +0000 UTC   2023-04-12 11:43:52.309190395 +0000 UTC
10.0.0.104   pool.ntp.org   2023-04-12 11:43:52.315811787 +0000 UTC   2023-04-12 11:43:52.307580784 +0000 UTC
10.0.0.105   pool.ntp.org   2023-04-12 11:43:52.297503558 +0000 UTC   2023-04-12 11:43:52.31071359 +0000 UTC
❯ talosctl --talosconfig=unwind-talosconfig -n 10.0.0.101,10.0.0.102,10.0.0.103,10.0.0.104,10.0.0.105 time -c time.cloudflare.com
NODE         NTP-SERVER            NODE-TIME                                 NTP-SERVER-TIME
10.0.0.101   time.cloudflare.com   2023-04-12 11:45:32.258062121 +0000 UTC   2023-04-12 11:45:32.249952965 +0000 UTC
10.0.0.102   time.cloudflare.com   2023-04-12 11:45:32.38238141 +0000 UTC    2023-04-12 11:45:32.245881097 +0000 UTC
10.0.0.103   time.cloudflare.com   2023-04-12 11:45:32.255538166 +0000 UTC   2023-04-12 11:45:32.249325211 +0000 UTC
10.0.0.104   time.cloudflare.com   2023-04-12 11:45:32.251933718 +0000 UTC   2023-04-12 11:45:32.245709705 +0000 UTC
10.0.0.105   time.cloudflare.com   2023-04-12 11:45:32.203954549 +0000 UTC   2023-04-12 11:45:32.249928454 +0000 UTC
```

### Environment

- Talos version: [`talosctl version --nodes <problematic nodes>`]

```sh
❯ talosctl --talosconfig=unwind-talosconfig -n 10.0.0.101,10.0.0.102,10.0.0.103,10.0.0.104,10.0.0.105 version
Client:
        Tag:         v1.3.6
        SHA:         undefined
        Built:
        Go version:  go1.20.2
        OS/Arch:     linux/amd64
Server:
        NODE:        10.0.0.101
        Tag:         v1.3.6
        SHA:         5b6176d8
        Built:
        Go version:  go1.19.7
        OS/Arch:     linux/amd64
        Enabled:     RBAC
        NODE:        10.0.0.104
        Tag:         v1.4.0-alpha.4
        SHA:         b8980817
        Built:
        Go version:  go1.20.2
        OS/Arch:     linux/amd64
        Enabled:     RBAC
        NODE:        10.0.0.103
        Tag:         v1.3.6
        SHA:         5b6176d8
        Built:
        Go version:  go1.19.7
        OS/Arch:     linux/amd64
        Enabled:     RBAC
        NODE:        10.0.0.102
        Tag:         v1.3.6
        SHA:         5b6176d8
        Built:
        Go version:  go1.19.7
        OS/Arch:     linux/amd64
        Enabled:     RBAC
        NODE:        10.0.0.105
        Tag:         v1.3.6
        SHA:         5b6176d8
        Built:
        Go version:  go1.19.7
        OS/Arch:     linux/amd64
        Enabled:     RBAC
```

- Kubernetes version: [`kubectl version --short`]

```sh
❯ kubectl version --short
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.26.3
Kustomize Version: v4.5.7
Server Version: v1.26.2
```

- Platform: Bare metal

### PR DESCRIPTION
**Pull request by smira**
Fixes #7080

The real bug was off-by-one in `log2i` implementation, other changes are cleanups as `x/sys/unix` package now contains all the constants we need.

### CODE REVIEW

File: internal/pkg/ntp/ntp.go [hunk 1]
```diff
@@ -394,13 +394,22 @@ func (syncer *Syncer) queryServer(server string) (*ntp.Response, error) {
 	return resp, err
 }
 
+// log2i returns 0 for v == 0 and v == 1.
+func log2i(v uint64) int {
+	if v == 0 {
+		return 0
+	}
+
+	return 63 - bits.LeadingZeros64(v)
```
Comment (smira) [inline internal/pkg/ntp/ntp.go:403]: this is the real fix, previous code was `s/63/64`.

the code in systemd for comparison: https://github.com/systemd/systemd/blob/6639ac474eb7a5325a72a3d7370492792dd00bc0/src/fundamental/logarithm.h#L39-L40

**Review by frezbo — APPROVED** [review]
+1 on not needing to maintain all these constants

**Comment by smira** [comment]
one more PR coming to help debugging things

**Pull request merged**